File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        result = ""
        n = len(s)

        for i in range(n):
            left = i
            right = i
            while left >= 0 and right < n and s[left] == s[right]:
                if len(result) < right-left+1:
                    result = s[left:right+1]
                left -= 1
                right += 1 

        for i in range(n):
            left = i
            right = left+1
            while left >= 0 and right < n and s[left] == s[right]:
                if len(result) < right-left+1:
                    result = s[left:right+1]
                left -= 1
                right += 1 

        return result

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {

    String maxString = "";
    String s;
    int n;

    public String longestPalindrome(String s) {

        this.s = s;
        this.n = s.length();

        for(int i = 0; i < n; i++){
            getMaxString(i, i+1);
        }

        for(int i = 0; i < n; i++){    
            getMaxString(i, i);
        }        

        return this.maxString;
    }
    private void getMaxString(int left, int right){
        while(left >= 0 && right < this.n && this.s.charAt(left) == this.s.charAt(right)){
            if(this.maxString.length() < right-left+1){
                this.maxString = this.s.substring(left, right+1);
            }
            left--;
            right++;
        }        
    }
}